README.md

```diff
@@ -1,7 +1,54 @@
-# ReefGuide Web API
+# ReefGuide Web API - Developer Guide
 
 A REST API to support Reef Guide (AIMS), built with Express, TypeScript, Zod and Prisma, deployable to AWS using CDK.
 
+## Table of Contents
+
+- [Features](#features)
+- [Quick Start](#quick-start)
+- [Prerequisites](#prerequisites)
+- [Local Development Setup](#local-development-setup)
+- [Testing Your Setup](#testing-your-setup)
+- [Development Workflow](#development-workflow)
+- [Database Management](#database-management)
+- [Testing](#testing)
+- [Project Structure](#project-structure)
+- [API Endpoints](#api-endpoints)
```

Comment:
This reminds me, I'd like to break this up into multiple files under `docs/` at some point, but doesn't need to be done in this PR.